### PR TITLE
Set tasks to always use CMDER_START

### LINKED ISSUE #772: CMDER_START should always be set to USERPROFILE unless explicitly set through /START parameter
- [x] CmderLauncher.cpp: set CMDER_START to USERPROFILE unless set through /START param.
- [ ] Conemu.xml: Always use CMDER_START for new tasks.

### PR DESCRIPTION
**Pull request by MartiUK**
CMDER_START should always be set as a result of either:
1. The user passes a directory to cmder.exe using `/START $DIR` OR
2. Sets CMDER_START as a default environment variable.
Fixes #772

### CODE REVIEW

**Comment by JanSchulz** [comment]
I don't understand the need for this. With https://github.com/cmderdev/cmder/pull/798, there are a few ways to startup:

* cmder here in windows explorer: will call `cmder.exe <path>` and `cmder.exe` will set `CMDER_START` which will be picked up by init.bat or profile.ps1 and change to that dir
* if a user specifies a `CMDER_START` this will take the same path as long as `cmder.exe` isn't called with a path/ `/START` (e.g. by "cmder here" or a commandline startup)
* if no path is supplied (e.g. by starting it up by a shortcut or plain `cmder.exe` in the shell), the current task definition will ensure that the shells are started in `USERPROFILE` which IMO are exactly the right dir to start in.

So IMO we are ok with the "start in the right dir" problem. Or what is still missing (-> whats the to be solved problem in #772)?

**Comment by JanSchulz** [comment]
Just to make that clear: the above is currently happening for PS and cmd, but not for bash tasks. To make that happen as well, this could be applied:

```diff
diff --git a/vendor/cmder.sh b/vendor/cmder.sh
index 4d1da3a..9e21e31 100644
--- a/vendor/cmder.sh
+++ b/vendor/cmder.sh
@@ -51,3 +51,8 @@ fi
 if [ -f "${HOME}/.bashrc" ] ; then
     . "${HOME}/.bashrc"
 fi
+
+if [ "$CMDER_START" == "" ] ; then
+    CMDER_START="$(cygpath -u "$CMDER_START")"
+    cd "${CMDER_START}"
+fi
```

**Comment by MartiUK** [comment]
I've reverted the config changes, although I'll keep the launcher changes as they resolve some issues where CMDER_START isn't set in some instances.

File: launcher/src/CmderLauncher.cpp [hunk 1]
```diff
@@ -153,17 +155,26 @@ void StartCmder(std::wstring path, bool is_single_mode)
...
-		//SendMessageTimeout(HWND_BROADCAST, WM_SETTINGCHANGE, 0, (LPARAM)"Environment", SMTO_ABORTIFHUNG, 5000, NULL);
-
-		// ...and once using UniCode (because Windows 8 likes it that way).
-		//SendMessageTimeout(HWND_BROADCAST, WM_SETTINGCHANGE, 0, (LPARAM) L"Environment", SMTO_ABORTIFHUNG, 5000, NULL);
+	}
+	else
+	{
+		wchar_t* homeProfile = 0;
+		SHGetKnownFolderPath(FOLDERID_Profile, 0, NULL, &homeProfile);
+		if (!SetEnvironmentVariable(L"CMDER_START", homeProfile)) {
```
Comment (JanSchulz) [inline launcher/src/CmderLauncher.cpp]: IM this is still a bad change: init.bat will now *always* set the path to Home if cmder isn't started with a path. This means that startup path in the conemu config is completely useless now :-(

This also means that you can't do what's asked for in https://github.com/cmderdev/cmder/issues/203

**Comment by JanSchulz** [comment]
See inline comments: IMO always setting `CMDER_START` is wrong. what is the usecase for that?

File: launcher/src/CmderLauncher.cpp (hunk 1 above)
Comment (MartiUK) [inline launcher/src/CmderLauncher.cpp]: Yep just realised :fearful:

**Pull request merged**